src/swapd/runtime.rs

```diff
@@ -867,6 +875,8 @@ impl Runtime {
                     connected_counterparty_node_id: get_node_id(&self.peer_service),
                     deal: self.deal.clone(),
                     monero_address_creation_height: self.monero_address_creation_height.clone(),
+                    bitcoin_checkpoint_height: self.syncer_state.bitcoin_height,
+                    monero_checkpoint_height: self.syncer_state.monero_height,
```

Comment:
Why are these set with the checkpoints and not when the respective funding addresses are created?

Reply:
when restoring a checkpoint, we don't need to scan back further than the last checkpoint for transactions that still matter for the swap state we're restoring to.

that being said, I think what we should be checkpointing is not the current block height, but subtract from it the finalisation threshold: In case of a reorg, we should expect up to our finalisation threshold of blocks to be removed, so a tx we're watching for (or txs related to an address we're watching) could be moved ahead by that many blocks, and we'd miss it when rewatching.